### ISSUE TITLE
Server start stuck when loading python model instantiating certain transformers model

### ISSUE THREAD

**Issue by AJHoeh** [triton-inference-server/server#4537]
**Description**
When startimg triton server with a python model which instantiates certain hugging face transformers models during `initialize()` the server gets stuck. Namely I tried it with [this checkpoint](https://huggingface.co/asahi417/tner-xlm-roberta-large-multiconer-multi/tree/main). Since it works if I use [this checkpoint](https://huggingface.co/Babelscape/wikineural-multilingual-ner) instead, I guess the Problem could be the model size? I also experienced a similiar Problem when using the aforementioned working checkpoint but also loading other models (onnx). Could this be related to memory/gpu settings of the docker container?

**Triton Information**
nvcr.io/nvidia/tritonserver:22.04-py3

**To Reproduce**
env:
```
conda create --name triton-transformers python=3.8.10
conda activate triton-transformers
conda install -c huggingface transformers==4.14.1 tokenizers==0.10.3
export PYTHONNOUSERSITE=True
comda-pack
```

model:

```
...
from transformers import AutoModelForTokenClassification,

class TritonPythonModel:
    def initialize(self, args):
        self.checkpoint = r"/checkpoints/tner-xlm-roberta-large-multiconer-multi/"
        self.ner_model = AutoModelForTokenClassification.from_pretrained(self.checkpoint)
        print('Initialized...')

    def execute(self, requests):
        ...
```

config:
```
backend: "python"
max_batch_size: 8
input [
  {
    name: "sequence"
    data_type: TYPE_STRING
    dims: [ -1, -1 ]
  }
]
output [
  {
    name: "entities"
    data_type: TYPE_STRING
    dims: [ -1, -1 ]
  }
]
parameters: [
  {
    key: "EXECUTION_ENV_PATH"
    value: { string_value: "/envs/triton-transformers.tar.gz"}
  }
]
```
**Expected behavior**
Triton Server should just start.

**Comment by AJHoeh** [comment]
@dyastremsky Thanks for the fast reply! Here is the log:
```
root@51db6547ef04:/opt/tritonserver# tritonserver --model-repository=/models --strict-mode=False --model-control-mode="explicit" --load-model=tner --log-verbose=1
I0622 17:38:36.227140 1325 shared_library.cc:108] OpenLibraryHandle: /opt/tritonserver/backends/pytorch/libtriton_pytorch.so
I0622 17:38:36.497222 1325 libtorch.cc:1381] TRITONBACKEND_Initialize: pytorch
I0622 17:38:36.497254 1325 libtorch.cc:1391] Triton TRITONBACKEND API version: 1.9
I0622 17:38:36.497258 1325 libtorch.cc:1397] 'pytorch' TRITONBACKEND API version: 1.9
I0622 17:38:36.497290 1325 shared_library.cc:108] OpenLibraryHandle: /opt/tritonserver/backends/tensorflow1/libtriton_tensorflow1.so
2022-06-22 17:38:36.607369: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
I0622 17:38:36.629888 1325 tensorflow.cc:2181] TRITONBACKEND_Initialize: tensorflow
I0622 17:38:36.629919 1325 tensorflow.cc:2191] Triton TRITONBACKEND API version: 1.9
I0622 17:38:36.629923 1325 tensorflow.cc:2197] 'tensorflow' TRITONBACKEND API version: 1.9
I0622 17:38:36.629926 1325 tensorflow.cc:2221] backend configuration:
{}
I0622 17:38:36.629946 1325 shared_library.cc:108] OpenLibraryHandle: /opt/tritonserver/backends/onnxruntime/libtriton_onnxruntime.so
I0622 17:38:36.632084 1325 onnxruntime.cc:2400] TRITONBACKEND_Initialize: onnxruntime
I0622 17:38:36.632107 1325 onnxruntime.cc:2410] Triton TRITONBACKEND API version: 1.9
I0622 17:38:36.632111 1325 onnxruntime.cc:2416] 'onnxruntime' TRITONBACKEND API version: 1.9
I0622 17:38:36.632113 1325 onnxruntime.cc:2446] backend configuration:
{}
I0622 17:38:36.638202 1325 shared_library.cc:108] OpenLibraryHandle: /opt/tritonserver/backends/openvino_2021_4/libtriton_openvino_2021_4.so
I0622 17:38:36.644235 1325 openvino.cc:1207] TRITONBACKEND_Initialize: openvino
I0622 17:38:36.644257 1325 openvino.cc:1217] Triton TRITONBACKEND API version: 1.9
I0622 17:38:36.644260 1325 openvino.cc:1223] 'openvino' TRITONBACKEND API version: 1.9
I0622 17:38:37.782719 1325 pinned_memory_manager.cc:240] Pinned memory pool is created at '0x303a50000' with size 268435456
I0622 17:38:37.782788 1325 cuda_memory_manager.cc:105] CUDA memory pool is created on device 0 with size 67108864
I0622 17:38:37.802293 1325 model_config_utils.cc:648] Server side auto-completed config: name: "tner"
max_batch_size: 8
input {
  name: "sequence"
  data_type: TYPE_STRING
  dims: -1
  dims: -1
}
output {
  name: "entities"
  data_type: TYPE_STRING
  dims: -1
  dims: -1
}
parameters {
  key: "EXECUTION_ENV_PATH"
  value {
    string_value: "/checkpoints/triton-transformers.tar.gz"
  }
}
backend: "python"

I0622 17:38:37.804991 1325 model_repository_manager.cc:1077] loading: tner:1
I0622 17:38:37.911608 1325 shared_library.cc:108] OpenLibraryHandle: /opt/tritonserver/backends/python/libtriton_python.so
I0622 17:38:37.916388 1325 python.cc:1830] 'python' TRITONBACKEND API version: 1.9
I0622 17:38:37.916431 1325 python.cc:1852] backend configuration:
{}
I0622 17:38:37.916438 1325 python.cc:1962] Shared memory configuration is shm-default-byte-size=67108864,shm-growth-byte-size=67108864,stub-timeout-seconds=30
I0622 17:38:37.916599 1325 python.cc:2010] TRITONBACKEND_ModelInitialize: tner (version 1)
I0622 17:38:37.917316 1325 model_config_utils.cc:1592] ModelConfig 64-bit fields:
I0622 17:38:37.917345 1325 model_config_utils.cc:1594]  ModelConfig::dynamic_batching::default_queue_policy::default_timeout_microseconds
I0622 17:38:37.917366 1325 model_config_utils.cc:1594]  ModelConfig::dynamic_batching::max_queue_delay_microseconds
I0622 17:38:37.917389 1325 model_config_utils.cc:1594]  ModelConfig::dynamic_batching::priority_queue_policy::value::default_timeout_microseconds
I0622 17:38:37.917393 1325 model_config_utils.cc:1594]  ModelConfig::ensemble_scheduling::step::model_version
I0622 17:38:37.917396 1325 model_config_utils.cc:1594]  ModelConfig::input::dims
I0622 17:38:37.917398 1325 model_config_utils.cc:1594]  ModelConfig::input::reshape::shape
I0622 17:38:37.917400 1325 model_config_utils.cc:1594]  ModelConfig::instance_group::secondary_devices::device_id
I0622 17:38:37.917403 1325 model_config_utils.cc:1594]  ModelConfig::model_warmup::inputs::value::dims
I0622 17:38:37.917405 1325 model_config_utils.cc:1594]  ModelConfig::optimization::cuda::graph_spec::graph_lower_bound::input::value::dim
I0622 17:38:37.917407 1325 model_config_utils.cc:1594]  ModelConfig::optimization::cuda::graph_spec::input::value::dim
I0622 17:38:37.917410 1325 model_config_utils.cc:1594]  ModelConfig::output::dims
I0622 17:38:37.917414 1325 model_config_utils.cc:1594]  ModelConfig::output::reshape::shape
I0622 17:38:37.917419 1325 model_config_utils.cc:1594]  ModelConfig::sequence_batching::direct::max_queue_delay_microseconds
I0622 17:38:37.917453 1325 model_config_utils.cc:1594]  ModelConfig::sequence_batching::max_sequence_idle_microseconds
I0622 17:38:37.917459 1325 model_config_utils.cc:1594]  ModelConfig::sequence_batching::oldest::max_queue_delay_microseconds
I0622 17:38:37.917464 1325 model_config_utils.cc:1594]  ModelConfig::sequence_batching::state::dims
I0622 17:38:37.917467 1325 model_config_utils.cc:1594]  ModelConfig::sequence_batching::state::initial_state::dims
I0622 17:38:37.917472 1325 model_config_utils.cc:1594]  ModelConfig::version_policy::specific::versions
I0622 17:38:37.917587 1325 python.cc:1769] Using Python execution env /checkpoints/triton-transformers.tar.gz
I0622 17:38:37.917643 1325 python.cc:2054] TRITONBACKEND_ModelInstanceInitialize: tner (GPU device 0)
I0622 17:38:37.917665 1325 backend_model_instance.cc:105] Creating instance tner on GPU 0 (8.6) using artifact ''
I0622 17:38:45.494068 1330 python.cc:630] Starting Python backend stub: source /tmp/python_env_d7TU1n/0/bin/activate && exec env LD_LIBRARY_PATH=/tmp/python_env_d7TU1n/0/lib:$LD_LIBRARY_PATH /opt/tritonserver/backends/python/triton_python_backend_stub /models/tner/1/model.py triton_python_backend_shm_region_1 67108864 67108864 1325 /opt/tritonserver/backends/python 336 tner
```

**Comment by AJHoeh** [comment]
@Tabrizian I did not compile a python backend stub myself since my understanding of the [nore here](https://github.com/triton-inference-server/python_backend#1-building-custom-python-backend-stub) is that this is only necessary if my conda environments python version differs from the one of the python backend stub shipped with the container?

**Comment by Tabrizian** [comment]
That's correct. If the Python version matches you don't need to compile the stub. @dyastremsky Could you please file a ticket so that we can investigate this further? Thanks.

**Comment by dyastremsky** [comment]
Yes, thanks for the quick sanity check Iman! I've filed a ticket for us to investigate.

**Comment by AJHoeh** [comment]
Thanks at both of you!

**Comment by wesselvdv** [comment]
I think I might be having the same issue, as I am seeing the same kind of logs on my end. The only difference is that I get one additional line after the `Starting Python backend stub...`

```
[4pdvGPU Warn(147:140049859440640:libvgpu.c:81)]: recursive dlsym : ompt_start_tool
```

**Comment by wesselvdv** [comment]
Yessir, that I did. Is there some additional logging I can further enable that would allow me to somehow detect that this segfault has occurred?

**Comment by Tabrizian** [comment]
@wesselvdv @AJHoeh  @kthui has fixed a number of issues related to Python backend hangs [here ](https://github.com/triton-inference-server/python_backend/commit/32974481c4875ad22bf318dcb2b80e316cdf5065)and [here](https://github.com/triton-inference-server/python_backend/commit/00468e5f9d4d61c4b2fabf941b87424291e81e36). They will be included in 23.02 release. If they still didn't resolve the hang, please open a new issue.

**Comment by Vincouux** [comment]
Hey all, I am having the exact same issue. After the line `Starting Python backend stub:`, it hangs. Running htop on the host machine shows no activity.

I am trying to run some preprocessing for my models in Triton using a Python model. Is there another way to achieve this ? Many thanks !
